service/models.py

```diff
@@ -182,21 +183,20 @@ def deserialize(self, data: dict):
             data (dict): A dictionary containing the resource data
         """
         try:
-            self.name = data["name"]
-            self.description = data["description"]
-            self.category = data["category"]
-            self.check_price(data["price"])
-            self.check_available(data["available"])
+            if "name" in data:
+                self.check_name(data["name"])
+            if "description" in data:
+                self.check_description(data["description"])
+            if "category" in data:
+                self.check_category(data["category"])
+            if "price" in data:
+                self.check_price(data["price"])
+            if "available" in data:
+                self.check_available(data["available"])
             if "rating" in data:
                 self.check_rating(data["rating"])
-            if "cumulative_ratings" in data:
-                self.check_cumulative_ratings(data["cumulative_ratings"])
             if "no_of_users_rated" in data:
                 self.check_no_of_users_rated(data["no_of_users_rated"])
-        except AttributeError as error:
```

Comment:
Shouldn't we maintain these `Errors` ?

Reply:
Added an error handler in `model.py/create` to check if all fields required are provided.

Reply:
I am sorry, I got confused a bit